bn-apps/cordapp-updates-distribution/corda-updates-transport/src/main/kotlin/net/corda/cordaupdates/transport/flows/GetResourceFlow.kt

```diff
@@ -0,0 +1,89 @@
+package net.corda.cordaupdates.transport.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import net.corda.cordaupdates.transport.flows.Utils.toCordaException
+import net.corda.cordaupdates.transport.flows.Utils.transporterFactory
+import net.corda.core.flows.FlowLogic
+import net.corda.core.flows.FlowSession
+import net.corda.core.flows.InitiatedBy
+import net.corda.core.flows.InitiatingFlow
+import net.corda.core.flows.StartableByRPC
+import net.corda.core.flows.StartableByService
+import net.corda.core.identity.CordaX500Name
+import net.corda.core.utilities.unwrap
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils.newSession
+import org.eclipse.aether.repository.RemoteRepository
+import org.eclipse.aether.spi.connector.transport.GetTask
+import org.eclipse.aether.spi.connector.transport.Transporter
+import java.net.URI
+import java.util.*
+
+/**
+ * Fetches a resource from a remote repository. Used internally by Maven Resolver.
+ *
+ * @param resourceLocation resource location provided by Maven Resolver
+ * @param repositoryHosterName x500Name of the repository hoster
+ */
+@StartableByService
+@StartableByRPC
+@InitiatingFlow
+class GetResourceFlow(private val resourceLocation : String, private val repositoryHosterName : String) : FlowLogic<ByteArray>() {
+    @Suspendable
+    override fun call() : ByteArray {
+        val repoHosterParty = serviceHub.identityService.wellKnownPartyFromX500Name(CordaX500Name.parse(repositoryHosterName))!!
```

Comment:
Don't throw an NPE here, throw a FlowException that indicates a typo in the repository name. Maybe in the exception you can use partiesFromName and list the possibilities, as a convenience to the admin who may have forgotten the exact incantation?

Reply:
Good catch. Changed.